qiskit/visualization/bloch.py

```diff
@@ -54,19 +54,36 @@
 import matplotlib.pyplot as plt
 from matplotlib.patches import FancyArrowPatch
 from mpl_toolkits.mplot3d import Axes3D, proj3d
+from mpl_toolkits.mplot3d.art3d import Patch3D
 
 from .utils import matplotlib_close_if_inline
 
 
-class Arrow3D(FancyArrowPatch):
+class Arrow3D(Patch3D, FancyArrowPatch):
     """Makes a fancy arrow"""
 
-    def __init__(self, xs, ys, zs, *args, **kwargs):
-        FancyArrowPatch.__init__(self, (0, 0), (0, 0), *args, **kwargs)
-        self._verts3d = xs, ys, zs
+    # Nasty hack around a poorly implemented deprecation warning in Matplotlib 3.5 that issues two
+    # deprecation warnings if an artist's module does not claim to be part of the below module.
+    # This revolves around the method `Patch3D.do_3d_projection(self, renderer=None)`.  The
+    # `renderer` argument has been deprecated since Matplotlib 3.4, but in 3.5 some internal calls
+    # during `Axes3D` display started calling the method.  If an artist does not have this module,
+    # then it issues a deprecation warning, and calls it by passing the `renderer` parameter as
+    # well, which consequently triggers another deprecation warning.  We should be able to remove
+    # this once 3.6 is the minimum supported version, because the deprecation period ends then.
```

Comment:
Fwiw, it'll be a while before we can make our matplotlib requirement `>=3.6`.

Reply:
Yeah I know - here it's just that I don't like putting in gross hacks without saying the condition that they can be removed again!